### PR TITLE
[wallet-ext] hide and unhide assets

### PR DESCRIPTION
**Pull request by Nikhil-Mysten**
## Description 

image
image
image

hide

unhide

## Test Plan 

Locally

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: apps/wallet/src/ui/app/components/nft-display/index.tsx [hunk 1]
```diff
@@ -18,13 +18,18 @@ const nftDisplayCardStyles = cva('flex flex-nowrap items-center h-full', {
 		animateHover: {
 			true: 'group',
 		},
+		listView: {
```
Comment (Jibz-Mysten) [inline apps/wallet/src/ui/app/components/nft-display/index.tsx:21]: it seems confusing to listView and wideView together?

File: apps/wallet/src/ui/app/pages/home/nfts/index.tsx [hunk 2]
```diff
@@ -58,31 +149,41 @@ function NftsPage() {
...
+									onClick={() => {
+										ampli.clickedCollectibleCard({
+											objectId,
+											collectibleType: type!,
+										});
+									}}
+									key={objectId}
+									className="no-underline relative"
+								>
+									<button
```
Comment (plam-ml) [inline apps/wallet/src/ui/app/pages/home/nfts/index.tsx:168]: we have a Button component, I would extend that instead if possible

**Comment by mystie711** [comment]
Seeing a lot of UI delta from Figma @Nikhil-Mysten.
Let me know if we should do a huddle and screen share walk through of how to use Figma. May be the tool is coming in your way.

File: apps/wallet/src/ui/app/pages/home/hidden-assets/index.tsx [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+						<div>
+							<strong>Sync error (data might be outdated)</strong>
+						</div>
+						<small>{(error as Error).message}</small>
+					</Alert>
+				) : null}
+				{hiddenNfts?.length ? (
+					<div className="flex flex-col w-full divide-y divide-solid divide-gray-40 divide-x-0 gap-2 mb-5">
+						{hiddenNfts
+							.filter((nft) => nft && internalHiddenAssetIds.includes(nft.objectId))
```
Comment (plam-ml) [inline apps/wallet/src/ui/app/pages/home/hidden-assets/index.tsx:159]: should we `useMemo` this?

**Comment by mamos-mysten** [comment]
Noticed if you hide all of your assets, you're left with this screen: 

image

It might be useful to show some messaging here indictaing that all assets are hidden. Or at the very least keep the original messaging ("No Assets Found") 

and maybe it could be something like: 

"No Assets Found"
(small text: View 17 hidden assets) 


cc @mystie711

File: apps/wallet/src/ui/app/pages/home/hidden-assets/index.tsx [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+				});
+				await set(HIDDEN_ASSET_IDS, newHiddenAssetIds);
+			};
+
+			const assetShownToast = async (objectId: string) => {
+				toast.custom(
+					(t) => (
+						<div
+							className="flex items-center justify-between gap-2 bg-white w-full shadow-notification border-solid border-gray-45 rounded-full px-3 py-2"
+							style={{
```
Comment (mamos-mysten) [inline apps/wallet/src/ui/app/pages/home/hidden-assets/index.tsx:108]: does this need to be in a `style` prop?

File: apps/wallet/src/ui/app/pages/home/nfts/index.tsx [hunk 5]
```diff
@@ -57,32 +163,46 @@ function NftsPage() {
...
-										borderRadius="xl"
-									/>
-								</ErrorBoundary>
-							</Link>
-						))}
+									}).toString()}`}
+									onClick={() => {
+										ampli.clickedCollectibleCard({
+											objectId,
+											collectibleType: type!,
```
Comment (mamos-mysten) [inline apps/wallet/src/ui/app/pages/home/nfts/index.tsx:177]: do we want to instrument hiding / showing of assets with amplitude?

File: apps/wallet/src/ui/app/shared/PageTitle.tsx [hunk 6]
```diff
@@ -42,7 +45,7 @@ function PageTitle({ title = '', back }: PageTitleProps) {
 					{title}
 				</Heading>
 			</div>
-			{back ? <div className="basis-8" /> : null}
+			{back ? <div className="basis-8">{after}</div> : after}
```
[the remark this answers is not in the document]
Comment (Nikhil-Mysten) [reply apps/wallet/src/ui/app/shared/PageTitle.tsx:48]: this is to add the assets menu and the new element above this is just being used for spacing

File: apps/wallet/src/ui/app/pages/home/hidden-assets/index.tsx (hunk 4 above)
Comment (Nikhil-Mysten) [reply apps/wallet/src/ui/app/pages/home/hidden-assets/index.tsx:108]: the custom toast doesn't seem to allow for the animations to be passed through the classnames unfortunately

File: apps/wallet/src/ui/app/shared/PageTitle.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (mamos-mysten) [reply apps/wallet/src/ui/app/shared/PageTitle.tsx:48]: ah yea Im a dummy. was looking for the `after` element, it's the three-dots

File: apps/wallet/src/ui/app/components/navigation/Navigation.module.scss [hunk 7]
```diff
@@ -19,6 +19,7 @@
 	transition: transform 300ms ease-in-out;
 	opacity: 1;
 	flex-direction: column;
+	z-index: 999;
```
[the remark this answers is not in the document]
Comment (Nikhil-Mysten) [reply apps/wallet/src/ui/app/components/navigation/Navigation.module.scss:22]: yes - clicking the navigation links was triggering the link to the nft details page without updating the z-index

[the remark this answers is not in the document]
Comment (Nikhil-Mysten) [reply apps/wallet/src/ui/app/components/navigation/Navigation.module.scss:22]: maxing out the navigation z-index seemed to make sense functionally

**Review by mamos-mysten — APPROVED** [review]

**Review by mystie711 — APPROVED** [review]
LGTM!

**Pull request merged**